src/skuld/vnode.clj

```
@@ -500,10 +500,12 @@
     (locking vnode
       (assert (not (zombie? vnode)))
       (if (= (:epoch msg) (epoch vnode))
-        (do
-          (->> (db/claim-task! (:db vnode) id i claim)
-               (queue/update! (:queue vnode)))
-          {})
+        (if-let [task (db/claim-task! (:db vnode) id i claim)]
+          (do
+            (queue/update! (:queue vnode) task)
+            {})
+          {:error (str "task " id " was not found on vnode " (full-id vnode))})
```

I cleaned this up to have the vnode return an error if it doesn't actually have the task instead of silently failing to update.